lib/transbank/sdk/onepay/models/item.rb

```diff
@@ -22,74 +19,86 @@ class Item
+      # @raise [ItemError] when description is null
       def description=(description)
-        unless description.is_a? String
-          raise ItemError "Description is not a String"
-        end
+        raise Errors::ItemError, "Description cannot be null" if description.nil?
         @description = description
       end
 
       # @param quantity [Integer] How many of units of [Item]
-      # @raise [ItemError] when given quantity is not an [Integer] or is less than zero
+      # @raise [ItemError] when given quantity is nil or less than zero.
       def quantity=(quantity)
-        unless quantity.is_a? Integer
-          raise ItemError "Quantity must be an Integer"
-        end
-
+        raise Errors::ItemError, "Quantity cannot be null" if quantity.nil?
         if quantity < 0
-          raise ItemError "Quantity cannot be less than zero"
+          raise Errors::ItemError, "Quantity cannot be less than zero"
         end
         @quantity = quantity
       end
 
       # @param amount [Integer] The value of each unit of [Item]
-      # @raise [ItemError] when amount is not an [Integer] or is less than zero.
+      # @raise [ItemError] when amount is nil or less than zero.
       def amount=(amount)
-        unless amount.is_a? Integer
-          raise ItemError "Amount must be an Integer"
-        end
-
+        raise Errors::ItemError, "Amount cannot be null" if amount.nil?
         if amount < 0
-          raise ItemError "Amount cannot be less than zero"
+          raise Errors::ItemError, "Amount cannot be less than zero"
         end
         @amount = amount
       end
 
       # @param additional_data [String] A string with whatever additional data the
       # Merchant might want to add
-      # @raise [ItemError] when additional_data is not a [String]
       def additional_data=(additional_data)
         additional_data = '' if additional_data.nil?
-        unless additional_data.is_a? String
-          raise ItemError 'Additional data must be a String'
-        end
         @additional_data = additional_data
       end
 
       # @param expire [Integer] Expiry for the Item
-      # @raise [ItemError] if value is not an [Integer]
       def expire=(expire)
-        unless expire.is_a? Integer
-          raise ItemError 'Expire must be an Integer'
-        end
+        expire = expire || 0
         @expire = expire
       end
+
+      # Return the total amount to pay for the Item, that is, amount * quantity
+      # @return [Numeric] the total amount to pay for the [Item]
+      def total
+        self.quantity * self.amount
+      end
+
+      # Override == to allow comparison between [Item]s
+      # @return [boolean] true if equal, false otherwise
+      def ==(another_item)
+        instance_variables.map! { |var| var.to_s.gsub!(/^@/, '') }
+          .reduce(true) do |result, current_instance_variable|
+            original_value = send(current_instance_variable)
```

Comment:
This is fragile. If I compare against something that's not the same class it'll raise an exception rather than returning `false`

Reply:
It will also explode if I add instance variables that don't happen to also be attributes

Reply:
I agree on that it'll raise when an instance variable is not an attribute, but if the classes aren't the same it shouldn't matter, since it only matters whether the object responds to "current_instance_variable"

Reply:
You are breaking the `==` contract. Raising an exception is not what should happen when you compare objects from different classes.